src/identity/IdentityProviderHttpHandler.ts

```diff
@@ -0,0 +1,121 @@
+import { parse } from 'url';
+import escapeStringRegexp from 'escape-string-regexp';
+import type { Provider } from 'oidc-provider';
+// This import probably looks very hacky and it is. Weak Cache is required to get the oidc
+// configuration, which, in turn, is needed to get the routes the provider is using.
+// It is probably very difficult to get the configuration because Panva does not want
+// it to be possible, but we must get the configuration to satisfy the needs of the CSS
+// architecture. See the "canHandle" method for an explantaion
+// eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
+// @ts-ignore
+import instance from 'oidc-provider/lib/helpers/weak_cache';
+import { getLoggerFor } from '../logging/LogUtil';
+import type { HttpHandlerInput } from '../server/HttpHandler';
+import { HttpHandler } from '../server/HttpHandler';
+import { NotImplementedHttpError } from '../util/errors/NotImplementedHttpError';
+import type { IdentityProviderFactory } from './IdentityProviderFactory';
+import type { IdpInteractionPolicyHttpHandler } from './interaction/IdpInteractionPolicyHttpHandler';
+
+/**
+ * Handles requests incoming the IdP and instantiates the IdP to
+ * be passed to all child IdpInteractionHttpHandlers
+ */
+export class IdentityProviderHttpHandler extends HttpHandler {
+  private readonly providerFactory: IdentityProviderFactory;
+  private provider: Provider | undefined;
+  private providerCreationPromise: Promise<Provider> | undefined;
+  private readonly interactionPolicyHttpHandler: IdpInteractionPolicyHttpHandler;
+  private readonly logger = getLoggerFor(this);
+
+  public constructor(
+    providerFactory: IdentityProviderFactory,
+    interactionPolicyHttpHandler: IdpInteractionPolicyHttpHandler,
+  ) {
+    super();
+    this.interactionPolicyHttpHandler = interactionPolicyHttpHandler;
+    this.providerFactory = providerFactory;
+  }
+
+  /**
+   * Create the provider or retrieve it if it has already been created
+   */
+  private async getGuaranteedProvider(): Promise<Provider> {
+    if (!this.provider) {
+      if (!this.providerCreationPromise) {
+        this.providerCreationPromise = this.providerFactory.createProvider(this.interactionPolicyHttpHandler);
+      }
+      try {
+        this.provider = await this.providerCreationPromise;
+      } catch (err: unknown) {
+        this.logger.error(err as string);
+        throw err;
+      }
+    }
+    return this.provider;
+  }
+
+  /**
+   * Checks whether or not the identity provider can handle a request. There are two ways
+   * an IdP can handle a request. Firstly, the provider itself can handle it, and secondly
+   * the Interaction Policy can handle it.
+   * NOTE: This method has a lot of hacks in it to get it to work with node-oidc-provider.
+   */
+  public async canHandle(input: HttpHandlerInput): Promise<void> {
```

Comment:
So I still don't understand why this is needed. Why don't we have a simple check:
- Is it `/.well-known/openid-configuration`?
- Is it `/idp/*`?

I.e., can we force Panva's stuff to only need `/idp/*` routes, if we stick to particular kinds of configurations?

Reply:
Done